### PR TITLE
Add Periphers for Heltec with GPS Module Details

### PR DESCRIPTION
**Pull request by 5kamisama5**
I will like to add more instructions for people to hook up their modules to for specific boards. This is an example of one which I tried to be as detailed as possible for newer members, this will keep the same question from being asked on the #help channel. I have another for CardKB and buzzer modules.

### CODE REVIEW

**Comment by 5kamisama5** [comment]
Hook up GPS module
Here's the picture I want to add

**Comment by 5kamisama5** [comment]
@rcarteraz I've made the changes suggested. Thanks for the suggestion. Hopefully it's better

**Comment by tekstrand** [comment]
@5kamisama5 you'll need to update the directories/files in this, the naming structure currently uses spaces and does not define a file type. Make sure you match the other files in the docs directory

**Comment by tekstrand** [comment]
hey @rcarteraz did you mean to assign this to me?

**Comment by 5kamisama5** [comment]
> @5kamisama5 you'll need to update the directories/files in this, the naming structure currently uses spaces and does not define a file type. Make sure you match the other files in the docs directory

Sorry, not exactly sure what mean. i did add .mdx to the change I just did, if that's what you mean.

**Comment by tekstrand** [comment]
Screenshot 2024-02-15 at 7 54 12 PM
See how your file names have spaces in them? They should match the other files in this repo. Same with the image. You can't use spaces in file names. You'll want to organize the new files into existing folders

**Comment by tekstrand** [comment]
@pdxlocations @rcarteraz should this be under hardware, or is this more of a project? I'm really not seeing any other content like this under `hardware`

**Comment by pdxlocations** [comment]
> @pdxlocations @rcarteraz should this be under hardware, or is this more of a project? I'm really not seeing any other content like this under `hardware`

I think hardware is where it is makes sense since it's specific to a Heltec peripheral (as far as pinouts). maybe heltec/peripherals/ similar to RAK https://meshtastic.org/docs/hardware/devices/rak/peripherals/

Not 100% sure of that statement above, but it would be good to come up with a standard for this type of integration.

**Comment by tekstrand** [comment]
I think peripherals makes sense 👍

**Comment by 5kamisama5** [comment]
Formatting is now similar to the RAK one.

**Comment by pdxlocations** [comment]
Could someone deploy a preview of this?  I'm not on the Vercel team.

**Comment by tekstrand** [comment]
Seems to be a `tab` html tag left open. I do think we need to get rid of the extra folder and just put this content under a `peripherals.mdx` though, to match how raspi does it.

**Comment by 5kamisama5** [comment]
> Seems to be a `tab` html tag left open. I do think we need to get rid of the extra folder and just put this content under a `peripherals.mdx` though, to match how raspi does it.

Updated to follow the directory of the Raspberry but I think people may think this applies to the other heltec devices. I don't know how to make it specific without my original directory that specified the V3. I also corrected the missing tab

**Comment by tekstrand** [comment]
Couple of things left
1. Need to update the filename on the image to not have spaces
2. I don't think the image is actually used in your document, if you want to have it visible you'll need to add it. [Docusaurus has some docs](https://docusaurus.io/docs/markdown-features/assets#images) on how to do that if you aren't familiar

**Comment by 5kamisama5** [comment]
I removed the spaces from the image name, and added the image to the document. Under wiring diagram @tekstrand 

Alt text for the image

**Comment by tekstrand** [comment]
The build is now failing because of a missing closing tag. If you click on `Details` to the right of `CI / build (pull request) you can see the details
```
(node:1927) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[INFO] [en] Creating an optimized production build...
[info] [webpackbar] Compiling Client
[info] [webpackbar] Compiling Server
[success] [webpackbar] Client: Compiled with some errors in 1.14m
Error:  Client bundle compiled with errors therefore further build is impossible.
Error: MDX compilation failed for file "/home/runner/work/meshtastic/meshtastic/docs/hardware/devices/heltec/peripherals.mdx"
Cause: Expected a closing tag for `</TabItem>` (11:1-19:4)
Details:
{
  "message": "Expected a closing tag for `</TabItem>` (11:1-19:4)",
  "name": "1:1",
  "reason": "Expected a closing tag for `</TabItem>` (11:1-19:4)",
  "ruleId": "end-tag-mismatch",
  "source": "mdast-util-mdx-jsx"
}
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

I think it's just missing the closing `</Tabs>` maybe?

**Comment by tekstrand** [comment]
Alternatively if you're running the project locally you can test the build on your machine by running `pnpm build`

**Comment by 5kamisama5** [comment]
finally found the missing </Tabs>

File: docs/hardware/devices/heltec/peripherals.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ 3. Solder a cable from the GND slot on the GPS module to GND pin on Heltec board.
+ 4. Solder a cable from left leg of NPN 2N2222 Transistor to VCC skit on GPS module.
+ 5. Solder a cable from Right leg of NPN 2N2222 Transistor to 3V/5V pin on Heltec board.
+ 6. Solder a cable from Middle leg of NPN 2N2222 Transistor to GPIO 48 of Heltec board. (You may choose your own GPIO pin)
+ 7. Go to Meshtastic app > Radio Configurations > Position
+ 8. Set GPS_RX_PIN to 48 (This will communicate to the TXD slot on the GPS)
+ 9. Set GPS_TX_PIN to 47 (This will communicate to the RXD slot on the GPS)
+ 10. Set PIN_GPS_EN to 46 (This will allow the meshtastic firmware to turn off the power on the GPS board with the user button of the Heltec Board by pressing it 3 times)
+
+## Wiring Diagram
```
Comment (tekstrand) [inline docs/hardware/devices/heltec/peripherals.mdx:59]: Sorry, just noticed one last thing, this, and the headings below it, should all be >3 as they are all children of GPS Module which is a 2

File: docs/hardware/devices/heltec/peripherals.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+## Conclusion
+
+By following this guide, you can enhance your Heltec ESP32 V3 board with valuable GPS capabilities and an RTC, while effectively managing power consumption. Whether for a hobby project or a more serious application, these additions significantly expand the potential of your device.
+
+</TabItem>
+<TabItem value="Keyboard">
+
+## Keyboard
+
+Commming Soon
```
Comment (tekstrand) [inline docs/hardware/devices/heltec/peripherals.mdx:80]: Extra m here and under buzzer

**Comment by 5kamisama5** [comment]
Thanks for the correction, this was such a learning experience. @tekstrand

**Review by tekstrand — APPROVED** [review]
Great job @5kamisama5 ! Thank you for sticking with it and making all of these changes. This will definitely enhance the documentation.

**Pull request merged**